src/library/scala/collection/mutable/HashTable.scala

```diff
@@ -409,7 +409,7 @@ private[collection] object HashTable {
 /** Class used internally.
   * @since 2.8
   */
-trait HashEntry [A, E <: HashEntry[A, E]] {
+private[mutable] trait HashEntry[A, E <: HashEntry[A, E]] {
```

Comment:
Hmm.. Even though I'd like to go ahead with this change, it makes life harder in scala-java8-compat which extracts the `Array[HashEntry]` from a HashMap and implements Java collections using it. I think the (only) way out is to write more code in Java (`private[mutable]` is public in bytecode, so it can be accessed in Java source).

Reply:
OK, I managed to work around it in scala-java8-compat. So I think we should keep the change here.